sympy/logic/boolalg.py

```diff
@@ -1048,6 +1048,9 @@ def _eval_simplify(self, **kwargs):
         # And and Or, we only simplify the partial expressions before using
         # patterns
         rv = self.func(*[a.simplify(**kwargs) for a in self.args])
+        if type(super()._eval_simplify(**kwargs)) == BooleanTrue:
```

Comment:
Why is this type check needed?

Reply:
If replaced directly, it'd write other simplifications in terms of And and Or.

Reply:
I see. A better way to write this would be

```py
rv = super()._eval_simplify(**kwargs)
if rv in [True, False]:
    return rv
```

Another option would be to add something to the pattern simplifications to handle this specific case in Xor, since otherwise we are just wasting a call to simplify_logic.

Reply:
You're right. I'll try it.